include/cache.h

```diff
@@ -13,45 +13,78 @@
 extern "C" {
 #endif
 
-#define _sys_cache_flush_sig(x) void (x)(vaddr_t virt, size_t size)
-
-#if defined(CONFIG_CACHE_FLUSHING)
-
-#if defined(CONFIG_ARCH_CACHE_FLUSH_DETECT)
-	typedef _sys_cache_flush_sig(_sys_cache_flush_t);
-	extern _sys_cache_flush_t *sys_cache_flush;
+/**
+ *
+ * @brief Flush d-cache lines to main memory
+ *
+ * No alignment is required for either addr or size, but since
+ * sys_cache_flush() iterates on the d-cache lines, a d-cache line alignment for
+ * both is optimal.
+ *
+ * The d-cache line size is specified either via the CONFIG_CACHE_LINE_SIZE
+ * kconfig option or it is detected at runtime.
+ *
+ * @param addr the pointer to start the multi-line flush
+ * @param size the number of bytes that are to be flushed
+ *
+ * @return N/A
+ */
+static inline void sys_cache_flush(void *addr, size_t size)
+{
+#ifdef CONFIG_CACHE_FLUSHING
+	arch_cache_flush(addr, size);
 #else
-	extern _sys_cache_flush_sig(sys_cache_flush);
+	ARG_UNUSED(addr);
+	ARG_UNUSED(size);
 #endif
```

Comment:
Better to do the following -- that way we don't need these superficial `ARG_UNUSED`s
```
if (IS_ENABLED(CONFIG_CACHE_FLUSHING)) {
    ...
}
```

Reply:
Instead of IS_ENABLED , I am just using ifdef CONFIG_CACHE_FLUSHING now and removed the else part where there were ARG_UNUSED statements. Let me know if this is fine. Thanks

Reply:
The problem with removing `ARG_UNUSED` is that, as far as C compiler (not preprocessor) is concerned, the `addr` and `size` are unused when `CONFIG_CACHE_FLUSHING=n` and it will display warnings.

Using `if (IS_ENABLED ...` instead of preprocessor ifdef ensures that C compiler is aware of its usage. This is, in fact, a recommended coding practice in the Linux kernel.

Reply:
Changed it to IS_ENABLED and added the declaration for functions arch_dcache_flush and arch_dcache_invd. Thanks